aiohttp/client.py

```diff
@@ -555,13 +555,18 @@ async def _request(
                             else InvalidUrlClientError
                         )
                         raise err_exc_cls(url)
-                    if auth and auth_from_url:
+                    # If `auth` was passed for an already authenticated URL,
+                    # disallow only if this is the initial URL; this is to avoid issues
+                    # with sketchy redirects that are not the caller's responsibility
+                    if not history and (auth and auth_from_url):
```

Comment:
Doesn't it make sense to just move this before the loop? Then save doing this check on redirects/retries completely.

Reply:
In principle it could, but `auth_from_url` is fetched in the loop which is why the check is there as well, otherwise we would end up with code duplication.